File: app/src/main/java/ga/forntoh/bableschool/ScoreSheetActivity.java
package ga.forntoh.bableschool;

import android.graphics.Color;
import android.graphics.PorterDuff;
import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.widget.TextView;

import com.alfianyusufabdullah.SPager;

import java.util.Objects;

import ga.forntoh.bableschool.fragments.TermScoreFragment;
import ga.forntoh.bableschool.fragments.YearGraphFragment;
import ga.forntoh.bableschool.fragments.YearScoreFragment;
import ga.forntoh.bableschool.utils.Utils;

public class ScoreSheetActivity extends BaseActivity {

    private TermScoreFragment firstTerm = new TermScoreFragment();
    private TermScoreFragment secondTerm = new TermScoreFragment();
    private TermScoreFragment thirdTerm = new TermScoreFragment();

    @SuppressWarnings("ConstantConditions")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_score_sheet);

        TextView toolbar_title = findViewById(R.id.title);
        TabLayout tabLayout = findViewById(R.id.tab_layout);

        toolbar_title.setText(R.string.my_score_sheet);

        setSupportActionBar(findViewById(R.id.toolbar));
        Objects.requireNonNull(getSupportActionBar()).setDisplayHomeAsUpEnabled(true);

        disableFlags(true);
        enableWhiteStatusBar();

        Bundle firstBundle = new Bundle();
        firstBundle.putInt("term", 1);
        firstTerm.setArguments(firstBundle);

        Bundle secondBundle = new Bundle();
        secondBundle.putInt("term", 2);
        secondTerm.setArguments(secondBundle);

        Bundle thirdBundle = new Bundle();
        thirdBundle.putInt("term", 3);
        thirdTerm.setArguments(thirdBundle);

        SPager sPager = findViewById(R.id.mainPage);
        sPager.initFragmentManager(getSupportFragmentManager());
        sPager.addPages("", new YearScoreFragment());
        sPager.addPages("1st Term", firstTerm);
        sPager.addPages("2nd Term", secondTerm);
        sPager.addPages("3rd Term", thirdTerm);
        sPager.addPages("", new YearGraphFragment());
        sPager.addTabLayout(tabLayout);
        sPager.build();

        Utils.setTabWidthAsWrapContent(tabLayout, tabLayout.getTabCount() - 1);
        tabLayout.getTabAt(tabLayout.getTabCount() - 1).setIcon(R.drawable.ic_multiline_chart);

        Utils.setTabWidthAsWrapContent(tabLayout, 0);
        tabLayout.getTabAt(0).setIcon(R.drawable.ic_class);

        sPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                if (position == 0 || position == tabLayout.getTabCount() - 1)
                    tabLayout.getTabAt(position).getIcon().setColorFilter(Color.parseColor("#2283FF"), PorterDuff.Mode.SRC_ATOP);
                else {
                    tabLayout.getTabAt(0).getIcon().setColorFilter(Color.parseColor("#C5C5C5"), PorterDuff.Mode.SRC_ATOP);
                    tabLayout.getTabAt(tabLayout.getTabCount() - 1).getIcon().setColorFilter(Color.parseColor("#C5C5C5"), PorterDuff.Mode.SRC_ATOP);
                }
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });
    }
}

File: app/src/main/java/ga/forntoh/bableschool/ui/profile/score/ScoreSheetActivity.kt
package ga.forntoh.bableschool.ui.profile.score

import android.graphics.Color
import android.graphics.PorterDuff
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.viewpager.widget.ViewPager
import ga.forntoh.bableschool.R
import ga.forntoh.bableschool.ui.BaseActivity
import ga.forntoh.bableschool.utils.enableWhiteStatusBar
import ga.forntoh.bableschool.utils.setTabWidthAsWrapContent
import kotlinx.android.synthetic.main.activity_score_sheet.*

class ScoreSheetActivity : BaseActivity(), ViewPager.OnPageChangeListener {

    private val firstTerm: TermScoreFragment by lazy {
        TermScoreFragment().apply {
            arguments = Bundle().apply { putInt("term", 1) }
        }
    }
    private val secondTerm: TermScoreFragment by lazy {
        TermScoreFragment().apply {
            arguments = Bundle().apply { putInt("term", 2) }
        }
    }
    private val thirdTerm: TermScoreFragment by lazy {
        TermScoreFragment().apply {
            arguments = Bundle().apply { putInt("term", 3) }
        }
    }

    private val toolbarTitle by lazy { findViewById<TextView>(R.id.title) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_score_sheet)

        toolbarTitle.setText(R.string.my_score_sheet)

        setSupportActionBar(toolbar as Toolbar?)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        disableFlags(true)
        enableWhiteStatusBar()

        main_pager.adapter = ScreenSlidePagerAdapter(supportFragmentManager)

        tab_layout.setupWithViewPager(main_pager)

        tab_layout.setTabWidthAsWrapContent(tab_layout.tabCount - 1)
        tab_layout.getTabAt(tab_layout.tabCount - 1)?.setIcon(R.drawable.ic_multiline_chart)

        tab_layout.setTabWidthAsWrapContent(0)
        tab_layout.getTabAt(0)?.setIcon(R.drawable.ic_class)

        main_pager.addOnPageChangeListener(this)
    }

    override fun onPageSelected(position: Int) {
        if (position == 0 || position == tab_layout.tabCount - 1)
            tab_layout.getTabAt(position)?.icon?.setColorFilter(Color.parseColor("#2283FF"), PorterDuff.Mode.SRC_ATOP)
        else {
            tab_layout.getTabAt(0)?.icon?.setColorFilter(Color.parseColor("#C5C5C5"), PorterDuff.Mode.SRC_ATOP)
            tab_layout.getTabAt(tab_layout.tabCount - 1)?.icon?.setColorFilter(Color.parseColor("#C5C5C5"), PorterDuff.Mode.SRC_ATOP)
        }
    }

    private inner class ScreenSlidePagerAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {
        override fun getCount(): Int = 5

        override fun getItem(position: Int) = when (position) {
            0 -> YearScoreFragment()
            1 -> firstTerm
            2 -> secondTerm
            3 -> thirdTerm
            else -> YearGraphFragment()
        }

        override fun getPageTitle(position: Int) = when (position) {
            1 -> "1st Term"
            2 -> "2nd Term"
            3 -> "3rd Term"
            else -> ""
        }
    }

    override fun onPageScrollStateChanged(state: Int) = Unit

    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) = Unit
}
